### PR TITLE
Fix linter warnings

### PR DESCRIPTION
**Pull request by ericgaspar**
- *Fix linter warnings*

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,19 +1,16 @@
-diaspora_ynh
-==========
+Diaspora for YunoHost
```
Comment (autra) [inline README.md]: Why did you remove the title?

File: README.md [hunk 2]
```diff
@@ -1,19 +1,16 @@
...
-
-[![Install diaspora with YunoHost](https://install-app.yunohost.org/install-with-yunohost.png)](https://install-app.yunohost.org/?app=diaspora)
+[![Integration level](https://dash.yunohost.org/integration/diaspora.svg)](https://dash.yunohost.org/appci/app/diaspora) ![](https://ci-apps.yunohost.org/ci/badges/diaspora.status.svg) ![](https://ci-apps.yunohost.org/ci/badges/diaspora.maintain.svg)[![Shipped version](https://img.shields.io/github/v/release/yunohost-apps/diaspora_ynh)](https://github.com/yunohost-apps/diaspora_ynh/releases)  
+[![Install Diaspora with YunoHost](https://install-app.yunohost.org/install-with-yunohost.svg)](https://install-app.yunohost.org/?app=diaspora)
 
 ## Overview
 
-> *This package allow you to install diaspora* quickly and simply on a YunoHost server.  
+> *This package allow you to install Diaspora quickly and simply on a YunoHost server.
```
Comment (autra) [inline README.md]: the official name is Diaspora* (with the star), so the correct fix is likely to escape the star. Thanks for the catch though :-)

File: check_process [hunk 3]
```diff
@@ -16,9 +16,6 @@
 		multi_instance=1
 		port_already_use=0
 		change_url=0 # not supported upstream
-;;; Levels
```
Comment (autra) [inline check_process:19]: I trust you on this one, I didn't really follow what the new best practices are :-D

File: scripts/backup [hunk 4]
```diff
@@ -18,7 +18,7 @@ ynh_abort_if_errors
 #=================================================
 # LOAD SETTINGS
 #=================================================
-ynh_script_progression --message="Loading installation settings..."
+ynh_print_info --message="Loading installation settings..."
```
Comment (autra) [inline scripts/backup:21]: `print_info` is the new `script_progression`?

File: scripts/backup [hunk 5]
```diff
@@ -30,18 +30,20 @@ final_path=$(ynh_app_setting_get --app=$app --key=final_path)
 # clean folder
 ynh_secure_remove --file="$final_path/backup"
 mkdir -p $final_path/backup
+
 #=================================================
 # BACKUP DIASPORA DATABASE
 #=================================================
-ynh_script_progression --message="Backup Diaspora DB..." --weight=10
```
Comment (autra) [inline scripts/backup:36]: why remove the log?

File: scripts/install [hunk 6]
```diff
@@ -179,6 +179,7 @@ ynh_app_setting_set $app unprotected_uris "/"
 # CREATE AN ADMIN
 #=================================================
 ynh_script_progression --message="Create admin..." --weight=1
+
```
Comment (autra) [inline scripts/install]: why adding a blank line here? (I don't really care, but you did that at several places, so I'm wondering why).

File: scripts/restore [hunk 7]
```diff
@@ -112,24 +115,27 @@ ynh_secure_remove --file=/tmp/diaspora.dump
 #=================================================
 # Bundle the ruby app
 #=================================================
-ynh_script_progression --message="Precompile assets..." --weight=200
+ynh_script_progression --message="Precompile assets..." --weight=20
```
Comment (autra) [inline scripts/restore]: mmmhhh on my machine, the 200 is *really* justified :-) What is your machine?

File: scripts/upgrade [hunk 8]
```diff
@@ -86,17 +90,29 @@ then
...
 
+#=================================================
+# ADVERTISE SERVICE IN ADMIN PANEL
+#=================================================
+yunohost service add $app.target \
+  --log $final_path/diaspora/log/production.log \
+        $final_path/diaspora/log/unicorn-stderr.log\
+        $final_path/diaspora/log/unicorn-stdout.log\
+        $final_path/diaspora/log/sidekiq.log\
+  --description "Diaspora service (unicorn web and sidekiq)"
```
Comment (autra) [inline scripts/upgrade:101]: are you sure this is necessary? It is already done at install time.

**Review by autra — COMMENTED** [review]
Thanks for this work!

I have some questions, probably because I don't follow example_ynh enough :-)

File: README.md [hunk 9]
```diff
@@ -1,19 +1,16 @@
-diaspora_ynh
-==========
+Diaspora for YunoHost
 
-[![Integration level](https://dash.yunohost.org/integration/diaspora.svg)](https://dash.yunohost.org/appci/app/diaspora) ![](https://ci-apps.yunohost.org/ci/badges/diaspora.status.svg) ![](https://ci-apps.yunohost.org/ci/badges/diaspora.maintain.svg)[![Shipped version](https://img.shields.io/github/v/release/yunohost-apps/diaspora_ynh)](https://github.com/yunohost-apps/diaspora_ynh/releases)
-
-[![Install diaspora with YunoHost](https://install-app.yunohost.org/install-with-yunohost.png)](https://install-app.yunohost.org/?app=diaspora)
+[![Integration level](https://dash.yunohost.org/integration/diaspora.svg)](https://dash.yunohost.org/appci/app/diaspora) ![](https://ci-apps.yunohost.org/ci/badges/diaspora.status.svg) ![](https://ci-apps.yunohost.org/ci/badges/diaspora.maintain.svg)[![Shipped version](https://img.shields.io/github/v/release/yunohost-apps/diaspora_ynh)](https://github.com/yunohost-apps/diaspora_ynh/releases)  
+[![Install Diaspora with YunoHost](https://install-app.yunohost.org/install-with-yunohost.svg)](https://install-app.yunohost.org/?app=diaspora)
```
Comment (autra) [inline README.md:4]: Oh, I get it. It's the svg logo. 

As a side note, it's often useful to split these kind of PR in several commits. That way, it is much more explicit for the reviewer. For instance, a commit called "Use svg logo" makes it immediately clear what's happening, whereas "fix linter" is much less explicit. (don't bother correcting it for this PR)

File: README.md (hunk 1 above)
Comment (ericgaspar) [reply README.md]: I forgot a `#` 🙄

File: check_process (hunk 3 above)
Comment (ericgaspar) [reply check_process:19]: Levels are no longer needed...

File: scripts/backup (hunk 4 above)
Comment (ericgaspar) [reply scripts/backup:21]: See explanation in the linter output warnings : https://ci-apps.yunohost.org/ci/job/525

File: scripts/restore (hunk 7 above)
Comment (ericgaspar) [reply scripts/restore]: The number is more a way to assess the proportion that the process will take than the actual time... You can se explanation in the linter report:  https://ci-apps.yunohost.org/ci/job/525

File: scripts/upgrade (hunk 8 above)
Comment (ericgaspar) [reply scripts/upgrade:101]: Cf linter warning: https://ci-apps.yunohost.org/ci/job/525

File: README.md (hunk 9 above)
Comment (ericgaspar) [reply README.md:4]: My apologizes. you're completely right ... I really have to discipline myself to do PR fragmented into specific commits.

File: scripts/backup (hunk 5 above)
Comment (ericgaspar) [reply scripts/backup:36]: The log?

File: README.md (hunk 2 above)
Comment (ericgaspar) [reply README.md]: I removed it because it is used for italics in Markdown... so *  doesn't show after Diaspora but end up italic text

Comment (autra) [reply README.md]: > so the correct fix is likely to escape the star

like this \\* (it will show up like this \*)

File: scripts/backup (hunk 4 above)
Comment (autra) [reply scripts/backup:21]: Ok that makes sense, thanks

File: scripts/backup (hunk 5 above)
Comment (autra) [reply scripts/backup:36]: The message, but I had my answer in the previous thread :-)

File: scripts/restore (hunk 7 above)
Comment (autra) [reply scripts/restore]: Ok but in this case, the other numbers becomes completely meaningless. For instance we still have above:

>  ynh_script_progression --message="Reinstalling rvm and ruby..." --weight=50

but the ruby installation time is actually a lot less than the precompiling of assets...

File: scripts/upgrade (hunk 8 above)
Comment (autra) [reply scripts/upgrade:101]: I guess this should be in ./create_services

File: scripts/restore (hunk 7 above)
Comment (autra) [reply scripts/restore]: Ok let's do:

- 20 for this one
- 10 for reinstalling rvm and ruby (btw this should not be rvm, because if I remember correctly we use rbenv now)
- 5 for all other steps > 1 

What do you think?

**Review by alexAubin — APPROVED** [review]
Quick review, randomly passing by, LGTM :+1:

File: scripts/upgrade (hunk 8 above)
Comment (autra) [reply scripts/upgrade:101]: @ericgaspar ping for the above comment ^ 😄 What do you think?

Comment (ericgaspar) [reply scripts/upgrade:101]: From what I understand the CI expects to see this in the *install*, *upgrade* and *restore* script. I'm not sure if it's a good idea to put it in the `creat_services` file ... but it's up to you

Comment (alexAubin) [reply scripts/upgrade:101]: Answering because i'm responsible for designing the warning eric mentions :sweat_smile:  : 

Eeeer yeah it could, though package_linter will like it more if you keep this instruction in the install/upgrade/restore script directly because it just does a "dirty" parsing/greping of stuff ... Though that brings again the legitimate question of "but duplicating code is blasphemy!" and indeed it is, but that's how we roll for now and hopefully someday™ in 2021 we'll have a new, better packaging format to reduce the madness involved in packaging apps and duplicating stuff all over the place ...

(Slightly related, I see in the `create_services` file the manipulation using `install` and `ynh_replace_string` - hopefully the new helper `ynh_add_config` from 4.1, which we hope to generalize a bit more in 4.2, should be helpful in reducing the complexity - though not sure why `ynh_add_systemd_config` wasn't used yet ?)

Comment (autra) [reply scripts/upgrade:101]: > Though that brings again the legitimate question of "but duplicating code is blasphemy!" and indeed it is, but that's how we roll for now

Yeah, yeah, I'm not going to stir the pot on this, I know better now :-D

> hopefully someday™ in 2021 we'll have a new, better packaging format

I'm out of the loop since the end of the confinement (for personal reasons), but if it'll allow us to avoid bash, I'm all for it :-)

> hopefully the new helper `ynh_add_config` from 4.1

oh, I'm gonna check this out, sounds interesting!

> though not sure why `ynh_add_systemd_config` wasn't used yet ?

I'm pretty sure it wasn't around when I wrote this. Or just I wasn't aware of this.

In short, let's not fight the linter, let's keep it like that. Thanks guys.

**Comment by autra** [comment]
Thanks @ericgaspar for the work! And thanks @alexAubin for passing by and speed things up :-)
